charmhelpers/contrib/openstack/context.py

```diff
@@ -1935,3 +1935,20 @@ def __call__(self):
         return {
             'openstack_release': ostack,
             'operating_system_release': osystem}
+
+
+class LogrotateContext(OSContextGenerator):
+    """Common context generator for logrotate."""
+
+    def __init__(self, location=None, interval=None, count=None):
```

Comment:
Should these values be `None`.  I suspect that, according to the example, the signature should be something like:

```python
    def __init__(self, location, interval=4, count=4):
```

This function also needs a docstring to describe the parameters and their meaning. e.g.

```python
    :param location: Absolute path for the log rotate config file
    :type location: str
    :param interval: The interval between rotations in x units (default 4)
    :type interval: int
    :param count: The number of log files to keep (default 4)
    :type count: int
```